substra/sdk/client.py

```diff
@@ -571,7 +612,8 @@ def add_compute_plan(self, data):
                 "test_data_sample_keys": list[str],
                 "traintuple_id": str,
                 "tag": str,
-            }]
+            }],
+            "metadata": list[str]
```

Comment:
you're missing the _update_metadata_field here. Also, we'll need to update the metadata field in each tuple of the compute plan.